### PR TITLE
Add pyfaidx, for etal/cnvkit#73.

### PR DESCRIPTION
**Pull request by mdshw5**
I've updated ngfrills.faidx.fasta_extract_regions to use pyfaidx. Haven't tested it yet but it should be mostly correct.

### CODE REVIEW

File: cnvlib/ngfrills/faidx.py [hunk 1]
```diff
@@ -2,107 +2,21 @@
...
         for chrom, rows in groupby(intervals, lambda cse: cse[0]):
-            # Seek to chrom offset in FASTA
-            try:
-                seq_len, offset, chars_per_line, bytes_per_line = index[chrom]
-            except KeyError:
-                raise ValueError("Sequence ID '" + chrom + "' is not in FASTA "
-                                 + "file " + fa_fname)
             logging.info("Extracting sequences from chromosome %s", chrom)
-            eol_size = bytes_per_line - chars_per_line  # Handle \n\r, \n
             for _chrom, start, end in rows:
```
Comment (mdshw5) [inline cnvlib/ngfrills/faidx.py]: @etal Are these start and end coordinates 0-based, 1-based?

Comment (etal) [inline cnvlib/ngfrills/faidx.py]: They are 0-based internally. I only convert to 1-based when writing to another file format.

**Comment by etal** [comment]
Awesome. I'll test it and merge this branch for the v0.8.0 release. Thanks!

**Comment by kyleabeauchamp** [comment]
FYI I recently upgraded the pyfaidx package (to 0.4.7.1; https://github.com/bioconda/bioconda-recipes/pull/1130) on bioconda, so that will be ready to go for the next cnvkit release.

**Comment by mdshw5** [comment]
Thanks, @kyleabeauchamp!

**Pull request merged**